### PR TITLE
RandLaNet update_probs label smoothing logic

### LINKED ISSUE #561: RandLaNet `update_probs` label smoothing logic is faulty
### Checklist

- [X] I have searched for [similar issues](https://github.com/isl-org/Open3D-ML/issues).
- [X] I have tested with the [latest development wheel](http://www.open3d.org/docs/latest/getting_started.html#development-version-pip).
- [X] I have checked the [release documentation](http://www.open3d.org/docs/release/) and the [latest documentation](http://www.open3d.org/docs/latest/) (for `master` branch).

### Describe the issue

The `RandLaNet.update_probs` is used to exponentially smooth the predicted results during inference. The best way of doing it should be that every time where there is a new prediction of a given point, the entire softmax distribution of that point is weighted between the old distribution and the new distribution. And in the end of the inference, the label of a given point is given by the argmax of this smoothed softmax distribution. However, based on the current behaviour, the final label is based on the argmax of the softmax distribution of the **last** prediction in stead of the smoothed prediction.  

```python3
    def update_probs(self, inputs, results, test_probs, test_labels):
        """Update test probabilities with probs from current tested patch.
        Args:
            inputs: input to the model.
            results: output of the model.
            test_probs: probabilities for whole pointcloud
            test_labels: ground truth for whole pointcloud.
        Returns:
            updated probabilities and labels
        """
        self.test_smooth = 0.95

        for b in range(results.size()[0]):

            result = torch.reshape(results[b], (-1, self.cfg.num_classes))
            probs = torch.nn.functional.softmax(result, dim=-1)
            probs = probs.cpu().data.numpy()
            labels = np.argmax(probs, 1)                                                    <--- HERE
            inds = inputs['data']['point_inds'][b]

            test_probs[inds] = self.test_smooth * test_probs[inds] + (
                1 - self.test_smooth) * probs
            test_labels[inds] = labels                                                         <--- HERE

        return test_probs, test_labels

```

### Steps to reproduce the bug

```python
It's a logical bug.
```

### Error message

_No response_

### Expected behavior

_No response_

### Open3D, Python and System information

```markdown
- Operating system: (e.g. OSX 10.15, Ubuntu 18.04, Windows 10 64-bit)
- Python version: (e.g. Python 3.8 / output from `import sys  print(sys.version)`)
- Open3D version: (output from python: `print(open3d.__version__)`)
- System type: (x84 / arm64 / apple-silicon / jetson / rpi)
- Is this remote workstation?: yes or no
- How did you install Open3D?: (e.g. pip, conda, build from source)
- Compiler version (if built from source): (e.g. gcc 7.5, clang 7.0)
```

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by mariusud**
Resolves #561

- - -
This change is [Reviewable](https://reviewable.io/reviews/isl-org/open3d-ml/569)

### CODE REVIEW

**Review by benjaminum — CHANGES_REQUESTED** [review]
@mariusud Can you give more details about the change to `update_probs()`?

> Reviewed 13 of 14 files at r1.
> *[Reviewable](https://reviewable.io/reviews/isl-org/open3d-ml/569#-:-NMcHa8t0-vkiNmNUoe4:bjvr6jd)* status: 13 of 14 files reviewed, 1 unresolved discussion (waiting on @mariusud)

___
*[`ml3d/torch/models/randlanet.py` line 450 at r1](https://reviewable.io/reviews/isl-org/open3d-ml/569#-NMcGQeJDQV05cNmcZ7O:-NMcGQeJDQV05cNmcZ7P:b-4uxl4i) ([raw file](https://github.com/isl-org/open3d-ml/blob/e89ea4bf6e341030fd3b3b179f5415c3a8c3b044/ml3d/torch/models/randlanet.py#L450)):*
> ```Python
> 
>         Returns:
>             updated probabilities and labels
> ```

Docstring for "Returns" needs to be updated

**Review by mariusud — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/isl-org/open3d-ml/569)* status: 4 of 14 files reviewed, 1 unresolved discussion (waiting on @benjaminum)

___
*[`ml3d/torch/models/randlanet.py` line 450 at r1](https://reviewable.io/reviews/isl-org/open3d-ml/569#-NMcGQeJDQV05cNmcZ7O:-NMdMeh777zc4pkhvrnW:bb6un1j) ([raw file](https://github.com/isl-org/open3d-ml/blob/e89ea4bf6e341030fd3b3b179f5415c3a8c3b044/ml3d/torch/models/randlanet.py#L450)):*
<details><summary><i>Previously, benjaminum (Benjamin Ummenhofer) wrote…</i></summary><blockquote>

Docstring for "Returns" needs to be updated
</blockquote></details>

Done

**Comment by benjaminum** [comment]
@mariusud the style check fails which can be easily fixed with `make apply-style`.
Any idea about the Ubuntu CI failing?

**Comment by mariusud** [comment]
Sorry, Don't know about the Ubuntu CI. Seems it is also failing on your dev branch?

**Comment by benjaminum** [comment]
@ssheorey Can you review the CI changes?

**Review by ssheorey — APPROVED** [review]
CI changes look good.

**Review by benjaminum — APPROVED** [review]
:lgtm: 

> *[Reviewable](https://reviewable.io/reviews/isl-org/open3d-ml/569#-:-NOhfYMVCtnb8mBrauoL:bg7igkv)* status: 3 of 14 files reviewed, 1 unresolved discussion
